src/sparseml/pytorch/utils/quantization/helpers.py

```diff
@@ -64,6 +67,214 @@
 )
 
 
+class QATWrapper(Module):
+    """
+    Wraps inputs and outputs of a Module or function with QuantStubs for
+    Quantization-Aware-Training (QAT)
+
+    :param forward_fn: function to be wrapped, should generally accept and return
+        torch Tensor(s)
+    :param num_inputs: number of inputs of the forward function to add a QuantStub
+        to. Will wrap the first num_inputs ordered inputs of the function. Default
+        is 1
+    :param num_outputs: number of outputs of the forward function to add a QuantStub
+        to. Will wrap the first num_inputs ordered outputs of the function. Default
+        is 1. Will also add a DeQuantStub for FP32 conversion if
+        torch.quantization.convert is invoked
+    :param input_qconfigs: QConfig to use for calibrating the input QuantStubs. Can
+        be a single QConfig that will be copied to each QuantStub or a list of one
+        QConfig for each input. Instead of a QConfig objects, the string 'asymmetric'
+        or 'symmetric' may be used to use default UINT8 asymmetric and symmetric
+        quantization respectively
+    :param output_qconfigs: QConfig to use for calibrating the output QuantStubs. Can
+        be a single QConfig that will be copied to each QuantStub or a list of one
+        QConfig for each output. Instead of a QConfig objects, the string 'asymmetric'
+        or 'symmetric' may be used to use default UINT8 asymmetric and symmetric
+        quantization respectively
+    """
+
+    @staticmethod
+    def from_module(module: Module) -> "QATWrapper":
+        """
+        :param module: torch Module to create a QATWrapper for
+        :return: QATWrapper object created using the given Module as the forward
+            function. Will attempt to find any other named parameter of the QATWrapper
+            constructor from the attributes of the given Module
+        """
+        qat_wrapper_param_names = [
+            "num_inputs",
+            "num_outputs",
+            "input_qconfigs",
+            "output_qconfigs",
+        ]
+        qat_wrapper_kwargs = {
```

Comment:
could we nest these one level deeper into a specifically named dictionary and then pull these attributes to ensure there's little chance for namespace collision on the original module? Mainly worried about the num_inputs and num_outputs as those can be fairly generic names

Reply:
good point, updating